modules/QnA/resources/qna.js

```diff
@@ -0,0 +1,356 @@
+				_this.getQnaContainer();
+				//qnaObject =  $(window['parent'].document.getElementById(embedPlayer.id )).parent().find( ".qnaInterface" );
+				qnaObject = _this.getQnaContainer().find(".qnaModuleBackground");
+				onVideoTogglePluginButton = $('.qna-on-video-btn');
+
+				// register to on click to change the icon of the toggle button
+                onVideoTogglePluginButton.on("click", function(){
+
+                    var openQnaContainer=!qnaObject.is(":visible");
+
+                    if (_this.getPlayer().layoutBuilder.fullScreenManager.isInFullScreen()) {
+                        _this.getPlayer().toggleFullscreen() ;
+                        openQnaContainer=true;
+                    }
+
+					_this.getQnaContainer();
+					if (openQnaContainer){
+						qnaObject.show();
+					} else {
+						qnaObject.hide();
+					}
+                    _this.changeVideoToggleIcon();
+				})
+
+				_this.updateUnreadBadge();
+			});
+
+			this.bind('onOpenFullScreen', function() {
+                qnaObject.hide();
+				_this.changeVideoToggleIcon();
+				if (!_this.getConfig( 'onPage' )) {
+					$( ".videoHolder, .mwPlayerContainer" ).css( "width", "100%" );	}
+			});
+			this.bind('onCloseFullScreen', function() {
+				_this.changeVideoToggleIcon();
+				if (!_this.getConfig( 'onPage' )){
+					setTimeout(function() {
+						$(".videoHolder, .mwPlayerContainer").css("width", _this.$qnaListContainer.width() - _this.getConfig('moduleWidth') + "px");
+					},0);
+				}
+			});
+		},
+
+		updateUnreadBadge: function(){
+			var _this = this;
+			// if its a number and is greater then 0 - show & update the badge
+			var num = _this.KQnaModule.getUnreadCount();
+			if (isNaN(num) || num <=0 ){
+				$('.qna-badge').hide();
+			}
+			else{
+				$('.qna-badge').text(num);
+				$('.qna-badge').show();
+			}
+		},
+
+		injectCssToPage: function(cssLink){
+			if (cssLink) {
+				cssLink = cssLink.toLowerCase().indexOf("http") === 0 ? cssLink : kWidget.getPath() + cssLink; // support external CSS links
+				$('head', window.parent.document).append('<link type="text/css" rel="stylesheet" href="' + cssLink + '"/>');
+			} else {
+				mw.log("Error: " + this.pluginName + " could not find CSS link");
+			}
+		},
+
+		// load the Q&A template to the div with qnaTargetId
+		getQnaContainer: function(){
+            var embedPlayer = this.getPlayer();
+			if (!this.$qnaListContainer) {
+				if ( this.getConfig( 'onPage' ) ) {
+					// Inject external CSS files
+					this.injectCssToPage(this.getConfig('qnaMainCssFileName'));
+					this.injectCssToPage(this.getConfig('qnaAnnouncementsCssFileName'));
+					this.injectCssToPage(this.getConfig('qnaFontsCssFileName'));
+					this.injectCssToPage(this.getConfig('qnaNanoCssFileName'));
+					this.injectCssToPage(this.getConfig('qnaThreadsListCssFileName'));
+
+					var iframeParent = window['parent'].document.getElementById(this.embedPlayer.id);
```

Comment:
every time we try to hit window.parent - must be under try&catch since sometimes you will be inside unfriendly iframe